Makefile

```diff
@@ -170,6 +170,7 @@ wheel:
 	# Copy libs to frontend/catalyst/lib
 	mkdir -p $(MK_DIR)/frontend/catalyst/lib/backend
 	cp $(RT_BUILD_DIR)/lib/librtd* $(MK_DIR)/frontend/catalyst/lib
+	cp $(RT_BUILD_DIR)/lib/libcatalyst_python_interpreter.* $(MK_DIR)/frontend/catalyst/lib
```

Comment:
Wouldn't this line fix the issue with `make wheel && python -m pip install dist/PennyLane_Catalyst-0.10.0.dev0-cp311-cp311-linux_x86_64.whl` now we package `libcatalyst_python_interpreter.*` in the wheels? 

We may need to update the RPATH property of the dependencies as follows to be able to define multiple paths for both development (make/editable installation) and deployment (wheels installation) scenarios :
```
set_property(TARGET my_target APPEND PROPERTY BUILD_RPATH "$ORIGIN;$ORIGIN/libs")
```